src/couch_file.erl

```diff
@@ -625,10 +627,12 @@ read_raw_iolist_int(#file{fd = Fd, pread_limit = Limit} = F, Pos, Len) ->
     case Pos + TotalBytes of
     Size when Size > F#file.eof + ?READ_AHEAD ->
         couch_stats:increment_counter([pread, exceed_eof]),
-        throw(read_beyond_eof);
+        {_Fd, Filepath} = get(couch_file_fd),
+        throw({read_beyond_eof, Filepath});
```

Comment:
This is wrong, isn't it? You are throwing `{read_beyond_eof, Filepath}` here, but then trying to catch `throw:read_beyond_eof` above, in `handle_call`.